src/StackExchange.Redis/VectorSetSimilaritySearchMessage.cs

```diff
@@ -0,0 +1,322 @@
+
+         if (!useFp32) count += values.Length; // {vector} in the VALUES case
+
+         if (useCheckAndSet) count++; // [CAS]
+         count += quantizationType switch
+         {
+             VectorQuantizationType.None or VectorQuantizationType.Binary => 1, // [NOQUANT] or [BIN]
+             VectorQuantizationType.Int8 => 0, // implicit
+             _ => throw new ArgumentOutOfRangeException(nameof(quantizationType)),
+         };
+
+         if (buildExplorationFactor.HasValue) count += 2; // [EF {build-exploration-factor}]
+         if (attributesJson is not null) count += 2; // [SETATTR {attributes}]
+         if (maxConnections.HasValue) count += 2; // [M {numlinks}]
+         return count;
+     }
+
+     protected override void WriteImpl(PhysicalConnection physical)
+     {
+         bool useFp32 = UseFp32; // snapshot to avoid race in debug scenarios
+         physical.WriteHeader(Command, GetArgCount(useFp32));
+         physical.Write(key);
+         if (reducedDimensions.HasValue)
+         {
+             physical.WriteBulkString("REDUCE"u8);
+             physical.WriteBulkString(reducedDimensions.GetValueOrDefault());
+         }
+         if (useFp32)
+         {
+             physical.WriteBulkString("FP32"u8);
+             physical.WriteBulkString(MemoryMarshal.AsBytes(values.Span));
+         }
+         else
+         {
+             physical.WriteBulkString("VALUES"u8);
+             physical.WriteBulkString(values.Length);
+             foreach (var val in values.Span)
+             {
+                 physical.WriteBulkString(val);
+             }
+         }
+         physical.WriteBulkString(element);
+         if (useCheckAndSet) physical.WriteBulkString("CAS"u8);
+
+         switch (quantizationType)
+         {
+             case VectorQuantizationType.Int8:
+                 break;
+             case VectorQuantizationType.None:
+                 physical.WriteBulkString("NOQUANT"u8);
+                 break;
+             case VectorQuantizationType.Binary:
+                 physical.WriteBulkString("BIN"u8);
+                 break;
+             default:
+                 throw new ArgumentOutOfRangeException(nameof(quantizationType));
+         }
+         if (buildExplorationFactor.HasValue)
+         {
+             physical.WriteBulkString("EF"u8);
+             physical.WriteBulkString(buildExplorationFactor.GetValueOrDefault());
+         }
+         if (attributesJson is not null)
+         {
+             physical.WriteBulkString("SETATTR"u8);
+             physical.WriteBulkString(attributesJson);
+         }
+         if (maxConnections.HasValue)
+         {
+             physical.WriteBulkString("M"u8);
+             physical.WriteBulkString(maxConnections.GetValueOrDefault());
+         }
+     }
+
+     public override int GetHashSlot(ServerSelectionStrategy serverSelectionStrategy)
+         => serverSelectionStrategy.HashSlot(key);
+         */
```

Comment:
Mean to leave to leave in for save for next writer PR?

Reply:
removed detritus